File: 01jupyter/nb/source/cs01/include/omp_parallel_num_threads.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char ** argv) {
  int nt = atoi(argv[1]);
  printf("before parallel\n");
#pragma omp parallel num_threads(nt)
  printf("in parallel\n");
  printf("after parallel\n");
  return 0;
}

File: 01jupyter/nb/source/02_parallel/include/omp_parallel_num_threads.f90
program omp_parallel_num_threads
  character(len=32) :: arg
  integer :: nt
  call get_command_argument(1, arg)
  read (arg, *) nt
  print "(a)", "before parallel"
  !$omp parallel num_threads(nt)
  print "(a)", "in parallel"
  !$omp end parallel
  print "(a)", "after parallel"
end program omp_parallel_num_threads
